app/src/script.js

```diff
@@ -89,10 +100,12 @@ function initializeState(state, tokenManagerContract, tokens, settings) {
 }
 
 const getAcceptedTokens = async (tokens, settings) => {
-  const promises = tokens.map(async tokenAddress => {
-    const token = await getTokenData(tokenAddress, settings)
-    return token
-  })
+  const promises = tokens
+    .filter(token => token != ETHER_TOKEN_FAKE_ADDRESS)
+    .map(async tokenAddress => {
+      const token = await getTokenData(tokenAddress, settings)
+      return token
+    })
```

Comment:
it's not necessary to await for the token data and then return it. `getAcceptedTokens` function is returning an array of promises so  instead you can do:  

```
.map(tokenAddress => getTokenData(tokenAddress, settings))

```